query/logicalplan/logicalplan.go

```diff
@@ -27,6 +27,16 @@ type LogicalPlan struct {
 	Aggregation *Aggregation
 }
 
+// IterOptions are a set of options for the TableReader Iterators
+// TODO: should we instead use the option pattern? Is that possible with the way the Iterator functions work?
```

Comment:
Not sure if it's possible. Would need to try locally. It would be very neat for sure.

Reply:
Yea I don't think it is without a serious rewrite